### PR TITLE
Feature - Add Issue/PR Templates

### PR DESCRIPTION
**Pull request by esizer**
## 👋 Introduction

As promised, I am adding templates for submitting issues and PRs. The following templates have been added:

- 🐛 Bug Report
- ♻️ Debt/Refactor
- ✏️ Documentation
- ✨ Feature
- ❓ Question
- 🧪 Tests
- 🛠️ Tooling
- 🌱 Pull Request

## 🕵️ Details

These should cover most use cases for submitting issues. I tried to cover all of the things I beleive are important when submitting different types of issues. I'd love for anyone to provide any additions and improvements they think could be made so we get well defined and easy to follow issues! 😄 

## 🧪 Testing

Honestly, I'm not 100% sure how to test these. What I usually do is open the project in VS Code and option click the file and select "Open in Preview" from the context menu. Or, you could open them in a MD editor if you have one.

If you have better ways of testing these, feel free to add them!

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/✨-feature-request.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Describe the feature that is being requested
+
+## 🕵️ Details
+
+Add any additional details that could assist with the development of the new feature.
+
+## 🎨 Design File
+
+Include a link to the design file (if it exists)
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/✨-feature-request.md:19]:
```suggestion
Include a link to the design file (if it exists).
```

File: .github/ISSUE_TEMPLATE/✨-feature-request.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+## 📸 Screenshot
+
+Add a screenshot of the new feature (if possible)
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/✨-feature-request.md:23]:
```suggestion
Add a screenshot of the new feature (if possible).
```

File: .github/ISSUE_TEMPLATE/✏️-documentation.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+name: "✏️ Documentation"
+about: Updates or addition to the documentation
+title: "✏️ Documentation"
+labels: documentation
+assignees: ''
+---
+
+## ✏️ Documentation
+
+Describe the update/addition that you are proposing
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/✏️-documentation.md:11]:
```suggestion
Describe the update/addition that you are proposing.
```

File: .github/ISSUE_TEMPLATE/✏️-documentation.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+## 🕵️ Details
+
+Add any additional details that could assist with the implementation of the new documentation.
+
+## 🙋‍♀️ Proposed Copy
+
+(optional) Do you have any copy you'd like to propose?
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/✏️-documentation.md:19]:
```suggestion
(optional) Do you have any copy that you would like to propose?
```

File: .github/ISSUE_TEMPLATE/♻️-debt-refactor.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+## 🌎 Localization
+
+(optional) Provide any new copy along with translations available.
+
+## ✅ Acceptance Criteria
+
+A set of assumptions which, when tested, verify that the debt was addressed and expected functionality has not been affected.
+
+- [ ] Criteria
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/♻️-debt-refactor.md:29]:
```suggestion
- [ ] Criteria 1
```

File: .github/ISSUE_TEMPLATE/♻️-debt-refactor.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+name: "♻️ Debt/Refactor"
+about: Something that needs to be improved or removed (without affecting expected functionality)
+title: "♻️ Debt/Refactor"
+labels: debt
+assignees: ''
+---
+
+## ♻️ Debt/Refactor
+
+Describe the problem with the existing implementation
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/♻️-debt-refactor.md:11]:
```suggestion
Describe the problem with the existing implementation.
```

File: .github/ISSUE_TEMPLATE/ 🐛-bug-report.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## 🙋‍♀️ Proposed Solution
+
+(optional) Do you have a proposed solution?
+
+## ✅ Acceptance Criteria
+
+A set of assumptions which, when tested, verify that the bug was addressed.
+
+- [ ] Criteria
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/ 🐛-bug-report.md:41]:
```suggestion
- [ ] Criteria 1
```

File: .github/ISSUE_TEMPLATE/ 🐛-bug-report.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+- **Browser**: 
+
+## 📋 Steps to Reproduce
+
+1. ...
+2. ...
+
+## 📸 Screenshot
+
+Add a screenshot if possible
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/ 🐛-bug-report.md:31]:
```suggestion
Add a screenshot (if possible).
```

File: .github/ISSUE_TEMPLATE/ 🐛-bug-report.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+name: "🐛 Bug Report"
+about: Report a bug to fix errors or improve UX
+title: "🐛 Bug Report"
+labels: bug
+assignees: ''
+---
+
+## 🐛 Bug
+
+Describe the issue you encountered
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/ 🐛-bug-report.md:11]:
```suggestion
Describe the issue you encountered.
```

File: .github/ISSUE_TEMPLATE/ 🐛-bug-report.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+## 🦋 Expected Behaviour
+
+What should be happening?
+
+## 🕵️ Details
+
+Add any additional details that could assist with troubleshooting/fixing the issue.
+
+- **Operating System**:
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/ 🐛-bug-report.md:21]:
```suggestion
- **Operating System**: Name and version (if possible).
```

File: .github/ISSUE_TEMPLATE/ 🐛-bug-report.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+## 🦋 Expected Behaviour
+
+What should be happening?
+
+## 🕵️ Details
+
+Add any additional details that could assist with troubleshooting/fixing the issue.
+
+- **Operating System**: 
+- **Browser**:
```
Comment (mnigh) [inline .github/ISSUE_TEMPLATE/ 🐛-bug-report.md:22]:
```suggestion
- **Browser**: Name and version (if possible).
```

**Review by mnigh — CHANGES_REQUESTED** [review]
**Excellent** 🏅 

Only a few grammar-related suggestions, great work!

**Comment by esizer** [comment]
> Only a few grammar-related suggestions, great work!

Thanks! A lot of typing from a bad typist results in a lot of mistakes 😜 I think I got committed all suggestions made 👍

**Review by mnigh — APPROVED** [review]
👍

**Comment by tristan-orourke** [comment]
👏

File: .github/ISSUE_TEMPLATE/ 🐛-bug-report.md [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+assignees: ''
+---
+
+## 🐛 Bug
+
+Describe the issue you encountered.
+
+## 🦋 Expected Behaviour
+
+What should be happening?
```
Comment (patcon) [inline .github/ISSUE_TEMPLATE/ 🐛-bug-report.md:15]: What about the standard "Expected Behaviour" vs "Observed Behaviour" contrast? I've always seen that, and it made sense to me as very clear :)

**Comment by patcon** [comment]
Still working through (no need to wait if you're feeling antsy), but couple general Q's for thought.

- using comments to hide instructions, and have default visible text for when it makes sense to provide none, so can just submit as-is if missing, and it's clear what's happened. e.g.,
```
## Screenshot

<!-- Add a screenshot (if possible). -->

None provided.
```

- Thoughts on always prompting for "Risks" or "Concerns" section whenever a solution is advocated? Coaxing us to be our own critical reviewer as an early step is something I've been advised on in the past (as the eternal optimist who's inclined to try to "sell" ppl on proposals first).

File: .github/ISSUE_TEMPLATE/❓-question.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
+---
+name: "❓ Question"
+about: Ask a question about a feature, proposal, or if you just need help!
+title: "❓ Question"
+labels: question
+assignees: ''
+---
+
+## ❓ Question
```
Comment (patcon) [inline .github/ISSUE_TEMPLATE/❓-question.md:9]: Is this a format we've ever used? What's the use-case?

(It seems like a template that's good for outside contributors, so I feel supportive of it! But if it's for the team, this seems like something we mostly use slack or daily scrum for, no? Should we be changing our behaviour to bring these into tickets?)

**Review by patcon — APPROVED** [review]
This looks great!

File: .github/ISSUE_TEMPLATE/❓-question.md (hunk 13 above)
Comment (esizer) [reply .github/ISSUE_TEMPLATE/❓-question.md:9]: Yup, we use it a lot :) Now, is a ticket the right place for it is a more philosophical topic! I think github discussions or slack threads are a more appropriate place for it. But for now, we are using it so I created a template to try to at least standardize it.

https://github.com/GCTC-NTGC/gc-digital-talent/labels/blocked%3A%20question

Comment (patcon) [reply .github/ISSUE_TEMPLATE/❓-question.md:9]: thanks!

And ooooh that's neat that linking to an issue renders it in a comment. Didn't realize that :/

Comment (esizer) [reply .github/ISSUE_TEMPLATE/❓-question.md:9]: ha! Neither did I until I did it :P

**Pull request merged**